### PR TITLE
(ListItem) Customize padding with pad prop

### LINKED ISSUE #1119: [FEATURE REQUEST] Make padding value between ListItem components customizable
If you go into the code of `ListItem`, you'll see that there is a component named `PadView` which adds spacing between the leftComponent, the title component & right component. It has a static value of 16.

This should be customizable and user should be allowed to add this pad value

### PR DESCRIPTION
**Pull request by Gregjarvez**
closes #1119

### CODE REVIEW

**Comment by Gregjarvez** [comment]
done !! ✔️

File: src/index.d.ts [hunk 1]
```diff
@@ -980,6 +980,7 @@ export interface ListItemProps {
   bottomDivider?: boolean;
   scaleProps?: ScaleProps;
   ViewComponent?: React.ComponentClass;
+  pad: number;
```
Comment (iRoachie) [inline src/index.d.ts]: `?` <- means it's optional

Comment (Gregjarvez) [reply src/index.d.ts]: 😀how did I miss that

**Comment by iRoachie** [comment]
LGTM!

**Pull request merged**